src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -689,8 +699,8 @@ def create_vm(cmd, vm_name, resource_group_name, image=None, size='Standard_DS1_
     if no_wait:
         return sdk_no_wait(no_wait, client.create_or_update,
                            resource_group_name, deployment_name, properties)
-    LongRunningOperation(cmd.cli_ctx)(sdk_no_wait(no_wait, client.create_or_update,
-                                                  resource_group_name, deployment_name, properties))
+    LongRunningOperation(cmd.cli_ctx)(client.create_or_update(resource_group_name, deployment_name, properties))
```

Comment:
sdk_no_wait has a no_wait parameter. no_wait is false here because of the above lines `if no_wait: return`. So it's a redundant code.